azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -95,13 +97,11 @@
 
       reports = validateProject(project, archive, file, prop);
```

Comment:
If there is validation error, is it safe to load and save the project?

Reply:
We want to return as many validation results as possible to users, including the errors found in loading project. If there is any error during validation or loading project, we will just return the report and not save it to DB. So I think this should be safe.